### PR TITLE
integrate with spellbook

### LINKED ISSUE #90: include scripts via a separate package, included with generator
- remove scripts from the generator and make a separate package for them
- have that package get installed via the generator by adding it as a dev dependency
- have `package.json` run scripts from that package

### PR DESCRIPTION
**Pull request by BrandonOCasey**
This will fix #90, #88, #85, #84, #74, and #59

### CODE REVIEW

**Comment by misteroneill** [comment]
This fixes #90

File: package.json [hunk 1]
```diff
@@ -42,7 +42,6 @@
   ],
   "dependencies": {
```
Comment (misteroneill) [inline package.json]: I think we're missing the spellbook dependency.

File: src/app/index.js [hunk 2]
```diff
@@ -140,15 +139,26 @@ export default yeoman.generators.Base.extend({
...
     let defaults = {
       bower: configs.hasOwnProperty('bower') ? !!configs.bower : true,
-      changelog: configs.hasOwnProperty('changelog') ? !!configs.changelog : true,
-      docs: configs.hasOwnProperty('docs') ? !!configs.docs : false,
       ghooks: configs.hasOwnProperty('ghooks') ? !!configs.ghooks : 'lint',
-      lang: configs.hasOwnProperty('lang') ? !!configs.lang : false,
       license: this._licenseDefault,
-      sass: configs.hasOwnProperty('sass') ? configs.sass : false,
+
+      noShimVideojs: configs.hasOwnProperty('noShimVideojs') ? !!configs.noShimVideojs : false,
```
Comment (misteroneill) [inline src/app/index.js]: I'm curious why we need this new option.

File: src/app/index.js [hunk 3]
```diff
@@ -618,6 +602,8 @@ export default yeoman.generators.Base.extend({
     if (this.options.hurry) {
       return;
     }
+    shelljs.exec('git add --all');
+    shelljs.exec('git commit -a -m "generator run"');
```
Comment (misteroneill) [inline src/app/index.js]: I'm not sure about this. I don't really like the idea of creating commits implicitly.

File: src/app/index.js [hunk 4]
```diff
@@ -382,15 +386,8 @@ export default yeoman.generators.Base.extend({
         templates: []
       },
       scripts: {
```
Comment (misteroneill) [inline src/app/index.js]: Can this object be removed entirely?

File: src/app/index.js (hunk 3 above)
Comment (BrandonOCasey) [inline src/app/index.js]: should we just do `git init` and leave it at that?

Comment (misteroneill) [inline src/app/index.js]: I think so. It's kind of a fine line between convenience and imposition and I feel like this crosses it just slightly.

For example, when running against an existing project, we don't want to be creating commits with files that may need manual cleanup.

File: package.json (hunk 1 above)
Comment (BrandonOCasey) [inline package.json]: The generator itself will not have to depend on `spellbook` unless we switch the build system to use it. The only problem with that would be the lack of support for running mocha tests, and even if we wanted to switch to qunit there is no good way to run qunit tests in node.

Comment (misteroneill) [inline package.json]: Yeah. I wasn't thinking! 😆

File: src/app/index.js (hunk 2 above)
Comment (misteroneill) [inline src/app/index.js]: Yeah, seems a bit too fiddly and I worry that users will get themselves into trouble.

File: package.json [hunk 5]
```diff
@@ -42,7 +42,6 @@
...
     "chalk": "^1.0.0",
-    "chg": "^0.3.2",
```
Comment (misteroneill) [inline package.json]: I actually need this in `master` now for an official CHANGELOG. Can this change be undone?

**Comment by misteroneill** [comment]
As discussed, we won't be supporting Windows for the immediate release of v3.0.0 but an eventual downstream release of videojs-spellbook should fix that without the need for another generator release.

**Comment by BrandonOCasey** [comment]
@misteroneill Ready for re-review :)

File: src/app/templates/_.gitignore [hunk 6]
```diff
@@ -30,6 +30,4 @@ node_modules/
 
 # Build-related directories
 dist/
-docs/api/
-es5/
-test/dist/
+
```
Comment (misteroneill) [inline src/app/templates/_.gitignore]: I feel like these should be ignored by Git...

Comment (BrandonOCasey) [inline src/app/templates/_.gitignore]: well `test/dist`, `docs/api` and `es5` will no longer exist in the root folder. All of them will be in `dist/test` or `dist/docs` or `dist/es5`

Comment (misteroneill) [inline src/app/templates/_.gitignore]: Ohhhh, right. Carry on!

**Comment by misteroneill** [comment]
A couple issues I noticed in testing:

- `localhost:9999/test/` no longer runs QUnit tests, which is fine, but the link from `index.html` in the root should be fixed to point, I guess, to `localhost:9876`.

**Comment by BrandonOCasey** [comment]
@misteroneill 
* The issues that you are seeing with: `localhost:9999/test/` may be due to uglify, running it locally with a test project causes the tests to work for me. What command where you running to start your server?
* Do we just want to cut out the scope at the front of the npm package?
* The css/sass failures appear to both be related to the banner, I will see if I can find out why.
* I will fix sb-release so that it can be run with npm version, npm run version, or as sb-release

**Comment by misteroneill** [comment]
- I started the server with `npm start`. Then `localhost:9999/test/` showed `NOT FOUND`. I don't recall if anything was logged in the console.
- The scope should just be in the package name. npm will install it in `@foo/bar` but all we have to do is put it in the name. Spellbook should probably "descope" project names if it gets it from `package.json`.
- Sounds good on `sb-release`! 👍

**Comment by misteroneill** [comment]
I changed the base branch because v3 will live on the `next` branch for a while as we maintain support for v2.

**Comment by BrandonOCasey** [comment]
ahh yeah banner is killing the build for you as you have a single quote in your name hmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm. The one setback of the command line shows its ugly head. let me see what I can do about it.

**Comment by BrandonOCasey** [comment]
@misteroneill Ready to be tested again!

**Comment by misteroneill** [comment]
Seems to be working great. Thanks @BrandonOCasey!

**Pull request merged**